app/controllers/sessions_controller.rb

```diff
@@ -13,19 +13,20 @@ def create
   end
 
   def destroy
-    log_out
+    log_out if logged_in?
     redirect_to root_url
   end
 
   private
 
   def login_success user
     log_in user
+    params[:session][:remember_me] == "1" ? remember(user) : forget(user)
     redirect_to user
   end
 
   def login_fail
-    flash[:danger] = t "error_login_message"
+    flash.now[:danger] = t "error_login_message"
```

Comment:
.now để khi đăng nhập chính xác sau khi đăng nhập sai, thông báo sẽ tự động tắt. Nếu không để .now thì sau khi log in chính xác sau lần log in lỗi, thông báo lỗi vẫn sẽ hiện lên.